### PR TITLE
magento/adobe-stock-integration#1146: Eliminate grid index and use as…

### LINKED ISSUE #1146: Move index table from MediaGalleryUi module to MediaGalleryIndexer
Now index table `media_gallery_asset_grid` for grid view is stored in the `MediaGalleryUi` module. UI module should contain presentation logic only and current approach lacks extensibility.

### Acceptance Criteria:
- index table is moved from `MediaGalleryUi` to `MediaGalleryIndexer` module
- independent indexer is created
- only one source is used for reindex (synchronization should not touch this table)

### PR DESCRIPTION
**Pull request by sidolov**
…sets table instead.

### Fixed Issues (if relevant)

1. magento/adobe-stock-integration#1146: Move index table from MediaGalleryUi module to MediaGalleryIndexer

### Manual testing scenarios (*)

1. ...
2. ...

### CODE REVIEW

File: MediaGalleryUi/view/adminhtml/ui_component/media_gallery_listing.xml [hunk 1]
```diff
@@ -183,20 +183,13 @@
...
         </column>
-        <column name="name">
+        <column name="title">
             <settings>
                 <label translate="true">Name</label>
                 <visible>false</visible>
                 <sortable>true</sortable>
             </settings>
         </column>
-        <column name="directory">
```
Comment (Nazar65) [inline MediaGalleryUi/view/adminhtml/ui_component/media_gallery_listing.xml:193]: I wonder if directoryTree filter will work afrer re mowing this column ?

**Review by Nazar65 — CHANGES_REQUESTED** [review]

**Comment by lenaorobei** [comment]
Looks like we need to remove `\Magento\AdobeStockImage\Model\SetLicensedInMediaGalleryGrid` with all the usages.

This table is also used -> `AdobeStockImage/etc/db_schema.xml` and `AdobeStockImage/etc/db_schema_whitelist.json`.

**Comment by sivaschenko** [comment]
@magento run Unit Tests

**Comment by sivaschenko** [comment]
@magento run Functional Tests EE

**Comment by sivaschenko** [comment]
Unit tests are green: http://10.234.235.163:8080/job/Unit-Tests/934/

**Comment by sivaschenko** [comment]
@magento run Functional Tests EE

**Comment by sivaschenko** [comment]
Functional Tests EE build: http://10.234.229.78:8080/job/Functional-Tests-EE/1876/

**Comment by sivaschenko** [comment]
Functional Tests EE build is green

**Comment by sivaschenko** [comment]
Tests: https://m2build-comm.devops.magento.com/job/MTS-API-Tests/8784/

**Pull request merged**